blocks/algorithms/__init__.py

```diff
@@ -180,7 +180,11 @@ class GradientDescent(DifferentiableCostMinimizer):
 
     """
     def __init__(self, step_rule=None, gradients=None, **kwargs):
-        super(GradientDescent, self).__init__(**kwargs)
+        if gradients and 'params' not in kwargs:
```

Comment:
if gradients:
        kwargs.setdefault("params", gradients.keys())
    super(GradientDescent, self).__init__(**kwargs)

would work and looks better than the current proposal.